mvc/src/main/java/com/interface21/webmvc/servlet/mvc/tobe/AnnotationHandlerMapping.java

```diff
@@ -14,16 +23,69 @@ public class AnnotationHandlerMapping {
     private final Object[] basePackage;
     private final Map<HandlerKey, HandlerExecution> handlerExecutions;
 
-    public AnnotationHandlerMapping(final Object... basePackage) {
+    public AnnotationHandlerMapping(final Object... basePackage) throws Exception {
         this.basePackage = basePackage;
         this.handlerExecutions = new HashMap<>();
+        initialize();
     }
 
-    public void initialize() {
+    public void initialize() throws Exception {
+        for (Class<?> controllerClass : findControllerClasses()) {
+            initializeHandlerExecutions(controllerClass);
+        }
+
         log.info("Initialized AnnotationHandlerMapping!");
     }
 
     public Object getHandler(final HttpServletRequest request) {
-        return null;
+        String url = request.getRequestURI();
+        RequestMethod requestMethod = RequestMethod.from(request.getMethod());
+
+        HandlerKey handlerKey = new HandlerKey(url, requestMethod);
+        return handlerExecutions.get(handlerKey);
+    }
+
+    private Set<Class<?>> findControllerClasses() {
+        Reflections reflections = new Reflections(basePackage);
+        return reflections.getTypesAnnotatedWith(Controller.class);
+    }
+
+    private void initializeHandlerExecutions(Class<?> controllerClass) throws Exception {
+        List<Method> mapperMethods = findMapperMethods(controllerClass);
+        Object controllerInstance = controllerClass.getDeclaredConstructor().newInstance();
+
+        for (Method mapperMethod : mapperMethods) {
+            addMappings(mapperMethod, controllerInstance);
+        }
+    }
+
+    private List<Method> findMapperMethods(Class<?> clazz) {
+        return Arrays.stream(clazz.getMethods())
+                .filter(method -> method.isAnnotationPresent(RequestMapping.class))
+                .toList();
+    }
+
+    private void addMappings(Method mapperMethod, Object controllerInstance) {
+        RequestMapping annotation = mapperMethod.getAnnotation(RequestMapping.class);
+        RequestMethod[] requestMethods = annotation.method();
+        String uri = annotation.value();
+
+        if (requestMethods.length == 0) {
+            requestMethods = RequestMethod.values();
+        }
```

Comment:
> @RequestMapping()에 method 설정이 되어 있지 않으면 모든 HTTP method를 지원해야 한다.

는 요구사항을 구현한 코드인 것 같아요! 구현은 매우 적절하지만 `requestMethods.length == 0` 가 위 요구사항의 의미를 적절히 반영하고 있는지는 생각해봐야 할 것 같아요. 아래와 같이 메서드를 분리해보는 것은 어떤가요? 

```suggestion
    private void addMappings(Method mapperMethod, Object controllerInstance) {
    RequestMapping annotation = mapperMethod.getAnnotation(RequestMapping.class);
    String uri = annotation.value();
    
    RequestMethod[] requestMethods = getRequestMethods(annotation);
    // ...
}

// 요청 메서드 목록을 가져오는 메서드
private RequestMethod[] getRequestMethods(RequestMapping annotation) {
    // 메서드가 명시되지 않으면 모든 HTTP 메서드를 기본값으로 사용
    RequestMethod[] requestMethods = annotation.method();
    if (requestMethods.length == 0) {
        return RequestMethod.values();
    }
    return requestMethods;
}
```